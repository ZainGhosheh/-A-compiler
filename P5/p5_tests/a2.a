A: custom {
    a: int = 5;
};
B: custom  {

};

main : () -> int {

    myA: A;
    a: immutable int;
    maybe a means 1 otherwise 2;
    a = 2;
    a++;
    fromconsole myA->a;
    fromconsole A;
    myA = A;
}
