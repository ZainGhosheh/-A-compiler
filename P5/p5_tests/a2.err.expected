FATAL [12,11]-[12,12]: Non-lval assignment
FATAL [13,5]-[13,6]: Non-lval assignment
FATAL [14,5]-[14,6]: Non-lval assignment
FATAL [16,17]-[16,18]: Attempt to assign user input to class
FATAL [17,5]-[17,8]: custom name assignment
Type Analysis Failed 
